ui/src/components/imquery/AddPropertyDialog.vue

```diff
@@ -63,29 +82,60 @@ watch(
   }
 );
 
+watch(
+  () => cloneDeep(selectedProperty.value),
+  newValue => {
+    if (isObjectHasKeys(selectedProperty.value)) {
+      whereOrMatch.value = buildProperty(selectedProperty.value as any);
+      if (isObjectHasKeys(whereOrMatch.value, ["typeOf", "where"])) {
+        editWhere.value = getEditWhere(whereOrMatch.value.where![0]!);
+        editWhereDMIri.value = getEditWhereDMIri(whereOrMatch.value.where![0]!) ?? (whereOrMatch.value as Match).typeOf?.["@id"];
```

Comment:
refactor this line to simplify conditional logic. Checks before adding as method inputs